hooks/useAppointment.tsx

```diff
@@ -188,9 +187,133 @@ export default function useAppointment(dispatchAppointment: React.Dispatch<IAppo
 		}
 	}
 
+	async function getAppointments(birthDate: string | null = null, nationalNumber: string | null = null) : Promise<object | undefined> {
+		Console.info("Getting appointments...");
+
+		let appointmentsURL = `
+			${Variables.DOMAINE_HTTP}/modules/Modules/QueueManagement/services/appointments.php?
+			id_project=${Variables.W_ID_PROJECT}
+			&serial=${Variables.SERIAL}
+		`;
+
+		if(!birthDate && !nationalNumber) {
+			Console.error("Error when trying to get appointments: birthDate and nationalNumber are both null or empty", { fileName: "useAppointment", functionName: "getAppointments", lineNumber: 200, });
+			dispatchError({
+				type: ERROR_ACTION_TYPE.SETERROR,
+				payload: {
+					hasError: true,
+					errorCode: ERROR_CODE.B400,
+					message: "Birth date and national number are both empty",
+				},
+			});
+			return undefined;
+		}
+
+		// ADD FILTERS
+		if(birthDate) 		appointmentsURL += `&birth_date=${birthDate}`;
+		if(nationalNumber) 	appointmentsURL += `&registre_national=${nationalNumber}`;
+		if(Variables.W_ID_SERVICE_FILTER && !isNaN(Variables.W_ID_SERVICE_FILTER)) 	appointmentsURL += `&id_service=${Variables.W_ID_SERVICE_FILTER}`;
+
+		// add time filter
+		const timeStart = new Date();
+		timeStart.setHours(timeStart.getHours() - 1);
```

Comment:
The interval allowed to perform the checkin can be given as a widget parameters (in minutes)